### PR TITLE
[RFC] doc: remove references to 'term' and other tty related options.

### PR DESCRIPTION
**Pull request by fmoralesc**
Re: https://github.com/neovim/neovim/issues/4218

### CODE REVIEW

File: runtime/doc/options.txt [hunk 1]
```diff
@@ -772,14 +772,11 @@ A jump table for the options with a short description can be found at |Q_op|.
 	putting a ":gui" command in the gvimrc file, before where the value
 	of 'background' is used (e.g., before ":syntax on").
 
-	For Windows the default is "dark".
-	For other systems "dark" is used when 'term' is "linux",
-	"screen.linux", "cygwin" or "putty", or $COLORFGBG suggests a dark
-	background.  Otherwise the default is "light".
+	For Windows the default is "dark". Otherwise the default is "light".
```
Comment (justinmk) [inline runtime/doc/options.txt]: I think we are planning to restore the COLORFGBG check.

Comment (fmoralesc) [inline runtime/doc/options.txt]: Yes, but we can restore the documentation on the logic at that point. When I checked how we could implement that when working on setting 'dark' as the default, I didn't find a way to do it, and I suspect that might take a while.

Comment (fmoralesc) [inline runtime/doc/options.txt]: Ah, sorry, I misunderstood. Yes, we could leave the note on $COLORFGBG.

File: runtime/doc/options.txt [hunk 2]
```diff
@@ -6552,7 +6539,8 @@ A jump table for the options with a short description can be found at |Q_op|.
 'ttybuiltin' 'tbi'	Removed. |vim-differences| {Nvim}
 
 				     *'ttyfast'* *'tf'* *'nottyfast'* *'notf'*
-'ttyfast' 'tf'		Removed. |vim-differences| {Nvim}
+'ttyfast' 'tf'		Deprecated. |vim-differences| {Nvim}
+			'set ttyfast' is ignored, 'set nottyfast' is an error.
```
Comment (justinmk) [inline runtime/doc/options.txt]: we're going to restore this. In fact we should do it right now :)

Comment (fmoralesc) [inline runtime/doc/options.txt]: OK, I just included it here because the documentation didn't reflect the status of the setting.

**Pull request merged**